### PR TITLE
IR-326: support get of image stream images of a manifest list

### PR DESCRIPTION
**Pull request by flavianmissi**

### CODE REVIEW

File: pkg/image/apiserver/registry/imagestreamimage/rest.go [hunk 1]
```diff
@@ -88,12 +90,35 @@ func (r *REST) Get(ctx context.Context, id string, options *metav1.GetOptions) (
...
+	} else if errors.IsNotFound(err) {
+		// couldn't find image in image stream.
+		// it could be that the image doesn't belong to this image stream,
+		// or the image is a sub-manifest of a manifest list which does belong to
+		// this image stream, so we try to find that before returning an error.
+		layers, inErr := r.imageStreamRegistry.GetImageStreamLayers(ctx, name, &metav1.GetOptions{})
+		if inErr != nil {
+			return nil, inErr
+		}
+		for _, image := range layers.Images {
```
Comment (dmage) [inline pkg/image/apiserver/registry/imagestreamimage/rest.go:106]: sub-manifests should be in the `Images`, so you don't need to iterate over `image.Manifests`

Comment (flavianmissi) [reply pkg/image/apiserver/registry/imagestreamimage/rest.go:106]: good point!

**Comment by flavianmissi** [comment]
putting this on hold until https://github.com/openshift/openshift-apiserver/pull/309 merges and this one is rebased.
/hold

**Comment by flavianmissi** [comment]
removed dependency on https://github.com/openshift/openshift-apiserver/pull/309 so we no longer have a dead lock.
/hold cancel

**Comment by xiuwang** [comment]
All scenarios behave like https://url.corp.redhat.com/IR-326 records.
/label qe-approved

**Pull request merged**